app/services/import_polygons.rb

```diff
@@ -0,0 +1,40 @@
+class ImportPolygons
+  attr_reader :location_type
+
+  def initialize(location_type:)
+    @location_type = location_type
+  end
+
+  def call
+    response = HTTParty.get(LOCATION_POLYGON_SETTINGS[location_type][:api])
+
+    response.fetch('features', []).each do |region_response|
+      region_name = region_response.dig('attributes', LOCATION_POLYGON_SETTINGS[location_type][:name_key]).downcase
+
+      next unless location_categories_include?(region_name)
+
+      geometry_rings = region_response.dig('geometry', 'rings')
+
+      # The first ring tends to contain far more points than subsequent rings.
+      # Boundary should be visualised to check how it should be used.
+      # If algolia searches by polygon are slow, these boundaries could be downsampled significantly.
+      points = []
+      geometry_rings[0].each do |point|
+        # API returns coords in an unconventional lng,lat order
+        points.push(*point.reverse.map { |coord| coord.round(13) })
+      end
+
+      LocationPolygon.find_or_create_by(name: region_name, location_type: location_type.to_s)
+                     .update(boundary: points)
+    end
+  end
+
+  private
+
+  def location_categories_include?(region_name)
+    location_type == :regions && DOWNCASE_REGIONS.include?(region_name) ||
+      location_type == :counties && DOWNCASE_COUNTIES.include?(region_name) ||
+      location_type == :local_authorities && DOWNCASE_BOROUGHS.include?(region_name) ||
```

Comment:
We could have the location type as `:london_boroughs` (or something else) if that is more revealing. Let's not have a mismatch between the `location_type` and what the data is referred to in the rest of the code base 👍

Reply:
I did think this too. I'll speak to Cesi and Ben about it.